src/control/ToolHandler.cpp

```diff
@@ -152,9 +154,15 @@ void ToolHandler::selectTool(ToolType type, bool fireToolChanged, bool triggered
 }
 
 void ToolHandler::fireToolChanged() {
-    if (listener) {
-        listener->toolChanged();
+    for (auto listener: this->toolChangeListeners) {
```

Comment:
This copies all listeners in `this->toolChangeListeners`. Was this intentional?
Better use:
```suggestion
    for (auto&& listener: this->toolChangeListeners) {
```

Reply:
No!!!! Thank you!!!!